pagopa/io-app
# feat: [IOBP-7] Add details screen for Discount type initiatives

## Short description
This PR introduces a refactoring of `InitiativeDetailsScreen` to support different initiatives types;

https://github.com/pagopa/io-app/assets/6160324/d7335838-cf5c-4ba2-ac0d-6c3840a7d3a9

## List of changes proposed in this pull request
- Added `InitiativeDetailsBaseScreenComponent`, which will be used as a base screen for every initiative type
- Refactored `InitiativeDetailsScreen` to use the new `InitiativeDetailsBaseScreenComponent`
- Added `InitiativeBonusCounter` which renders a bonus counters
- Added required locales
- Added tests for new components and older ones (where missing)

## How to test
Tests should solve correctly.
To test on the simulator with `io-dev-server`, follow the instruction in [this PR](https://github.com/pagopa/io-dev-api-server/pull/250) and configure the server to return some `DISCOUNT` type initiative. Now, from the wallet, navigate to the initiative details and check if everything works.
